RFCs/0000-expand-new-task.md

```diff
@@ -0,0 +1,164 @@
+
+- When one or more implementation files are provided, PDK will only generate a `tasks/<taskname>.json` file for the new task. The generated metadata will contain an `"implementations"` key with each given implementation file listed.
+
+- If the `--language` option is provided at the same time as one or more `<implementation>` arguments, PDK should exit with an appropriate error message.
+
+#### Examples
+
+The following example illustrates how a user might use PDK to author a new "cross-platform" task with Windows- and Linux-specific implementations:
+
+```bash
+$ pdk new task --private --language=bash sql_linux
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_linux.sh' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_linux.json' from template.
+
+$ pdk new task --private --language=powershell sql_windows
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_windows.ps1' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_windows.json' from template.
+
+$ pdk new task sql tasks/sql_linux.sh tasks/sql_windows.ps1
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql.json' from template.
+pdk (INFO): Added implementation 'sql_linux.sh' to '/Users/jesse/sandbox/pdk/testmod/tasks/sql.json'
+pdk (INFO): Added implementation 'sql_windows.rb' to '/Users/jesse/sandbox/pdk/testmod/tasks/sql.json'
+```
+
+Given the following task implementation script already exists at `tasks/init.rb`:
+
+```ruby
+#!/usr/bin/env ruby
+require 'json'
+
+params = JSON.parse(STDIN.read)
+action = params['action'] || params['_task']
+if ['start',  'stop'].include?(action)
+  `systemctl #{params['_task']} #{params['service']}`
+end
+```
+
+The following example illustrates how a user might use PDK to author a new set of "shared-implementation" tasks
+
+```bash
+$ pdk new task start tasks/init.rb
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/start.json' from template.
+pdk (INFO): Added implementation 'init.rb' to '/Users/jesse/sandbox/pdk/testmod/tasks/start.json'
+
+$ pdk new task stop tasks/init.rb
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/stop.json' from template.
+pdk (INFO): Added implementation 'init.rb' to '/Users/jesse/sandbox/pdk/testmod/tasks/stop.json'
+```
+
+## Unresolved Questions
+
+- Should `--private` and/or `--remote` be disallowed in combination with `<implementations>`?
+
+- Is there a way to infer implementation "requirements" from the given implementation arguments? E.g. can we safely infer a `shell` requirement for an implementation file ending in `.sh`?
+
+## Future Considerations
+
+- As Bolt continues to iterate and add functionality we may need to add additional options and flags to the task generators.
+
+- We should consider ways that PDK could help authors define their task's parameters.
```

Comment:
Should this RFC cover interactive use? e.g. what would an interview process look like?

Reply:
I suggest that be a followup RFC.